Fix not being able to reopen signals panel immediately after connecting a signal

Hilderin opened the pull request:
Fixes #92996 

The issue was that `last_valid_node` was being set to null when `p_node` was null, causing the loss of the last valid node. After a signal connection, `restore_last_valid_node` is called, but `last_valid_node` no longer contained the node being edited.

Hilderin committed:
Fix Can't reopen signals panel immediately after connecting a signal #92996

KoBeWi commented:
Something is wrong. The Node dock shows the connecting node, but the inspector does not.

https://github.com/godotengine/godot/assets/2223172/24c14a63-4e21-43bf-81cd-dbcf4d7ca96d

Hilderin commented:
I changed a couple of things because it's was weird that the NodeDock and Inspector were not working the same way.

I changed the fix that was made from #84530 simply by add the line `NodeDock::get_singleton()->set_node(nullptr);` in `scene_tree_dock.cpp` when the node is deleted.
Also, I added a `previous_item` method in `EditorNode`  so the `_add_callback` could select back the last node.
That allowed me to remove everything managing the last valid node in NodeDock.

KoBeWi commented on editor/editor_node.cpp:
```diff
@@ -2304,6 +2304,12 @@ void EditorNode::push_item(Object *p_object, const String &p_property, bool p_in
 	_edit_current();
 }
 
+void EditorNode::previous_item() {
```
```suggestion
void EditorNode::edit_previous_item() {
```
Make more sense I think.

Hilderin replied:
Done!

KoBeWi commented on editor/plugins/script_editor_plugin.cpp:
```diff
@@ -2856,6 +2856,8 @@ void ScriptEditor::_add_callback(Object *p_obj, const String &p_function, const
 		break;
 	}
+
+	EditorNode::get_singleton()->previous_item();
```
This might deserve a comment, as it makes an assumption about editing history.

Hilderin replied:
Done!

KoBeWi commented:
Note that `restore_last_valid_node()` was added in #54071 to fix the same issue, but seems like it no longer works correctly for some reason.

akien-mga commented:
Thanks!
